src/xbot/common/controls/sensors/JoystickButtonManager.java

```diff
@@ -44,14 +52,28 @@ public void addAnalogButton(AnalogHIDDescription desc) {
     }
 
     public AdvancedJoystickButton getifAvailable(int buttonNumber) {
+        
+        if (buttonNumber < 1 || buttonNumber > maxButtons) {
+            String message = "button " + buttonNumber + " is out of range!";
+            return getMockButton(message);
+        }
+        
         if (buttonMap.containsKey(buttonNumber)) {
             return buttonMap.remove(buttonNumber);
         } else {
             // Warn people that terrible things are happening, then return a null button.
-            log.error("button " + buttonNumber + " is already used! Cannot be used twice!");
-            return null;
+            String message = "button " + buttonNumber + " is already used! Cannot be used twice!";
+            return getMockButton(message);
         }
     }
+    
+    private AdvancedJoystickButton getMockButton(String message) {
```

Comment:
I don't think that we want a method called `getMockButton` to log as well and throw exceptions. I think it would be better to either:
- Log and throw outside of the method instead
- Rename it to `handleBadButton` (or similar)